src/PABLO/Octant.cpp

```diff
@@ -1637,6 +1836,166 @@ array<int64_t,3> Octant::getPeriodicCoord(uint8_t iface) const {
 
 };
 
+/*! Get the coordinates of the octant shifted through node inode and
+ * near the opposite periodic boundary (i.e. the coordinates considering this octant
+ * as a ghost periodic octant).
+ * The directions through the octant is periodic has to be passed as boolean flags.
+ * \param[in] inode Local index of the node target.
+ * \return Coordinates of octant considered as periodic ghost out of the logical domain.
+ */
+array<int64_t,3> Octant::getNodePeriodicCoord(uint8_t inode) const {
+    array<int64_t,3> coord;
+    coord[0] = this->m_x;
+    coord[1] = this->m_y;
+    coord[2] = this->m_z;
+    int64_t dh = this->getLogicalSize();
+    int64_t maxLength = int64_t(1)<<TreeConstants::MAX_LEVEL;
+
+    uint8_t iface1 = sm_treeConstants[m_dim].nodeFace[inode][0];
+    uint8_t iface2 = sm_treeConstants[m_dim].nodeFace[inode][1];
+    uint8_t iface3 = sm_treeConstants[m_dim].nodeFace[inode][m_dim-1];
+
+    switch (iface1) {
+    case 0 :
+    {
+        if (m_info[0]){
+            coord[0] = maxLength;
+        }
+    }
+    break;
+    case 1 :
+    {
+        if (m_info[1]){
+            coord[0]  = -dh;
+        }
+    }
+    break;
+    }
+
+    switch (iface2) {
+    case 2 :
+    {
+        if (m_info[2]){
+            coord[1]  = maxLength;
+        }
+    }
+    break;
+    case 3 :
+    {
+        if (m_info[3]){
+            coord[1] = -dh;
+        }
+    }
+    break;
+    }
+
+    switch (iface3) {
+    case 4 :
+    {
+        if (m_info[4]){
+            coord[2] = maxLength;
+        }
+    }
+    break;
+    case 5 :
+    {
+        if (m_info[5]){
+            coord[2] = -dh;
+        }
+    }
+    break;
+    }
+
+    return coord;
+
+};
+
+/*! Get the coordinates of the octant shifted through edge iedge and
+ * near the opposite periodic boundary (i.e. the coordinates considering this octant
+ * as a ghost periodic octant).
+ * The face through the octant is periodic has to be passed as boolean flags.
```

Comment:
For this comments it seems there should be an additional argument that tells which are the periodic faces. (same applies for the other get*PeriodicCoord() and compute*PeriodicOctant functions).